Collection save api

mcmuglim opened the pull request:
@tmillett R 👍
@adzellman O

tmillett commented on www/CQMContext.js:
```diff
@@ -311,4 +323,94 @@ CQMContext.prototype._getEntitlementInfo = function(entityNames, successCallback
+
+    var success = successCallback && function(json) {
+        var returnEntity = new Entity(json.rawEntity)
+        successCallback([returnEntity, json.downloadPct]);
```
hm so the only way someone can know if the call was successful is to wait for progress updates? So if this method was called and then for some reason it was queued (like another saveEntity call was sent immediately after) the callback may not be called for a while?

tmillett replied:
The workaround is they could call getEntityStatus if they wanted right?

mcmuglim replied:
Good point. We could return something to the successCallback earlier. Like when an operation was created successfully and give it a 0% to start with. 
Or/And return an extra field, like what we returned for getEntityStatus -- ProgressState.

mttmllns commented on www/CQMContext.js:
```diff
@@ -52,6 +52,18 @@ var Entity = function(newEntity) {
 /**
+ * This class represents entity download progress state
+ */
+var ProgressState = function(state) {
```
Do we use this?  I don't see `new ProgressState()` called anywhere.

mttmllns commented on www/CQMContext.js:
```diff
@@ -311,4 +323,94 @@ CQMContext.prototype._getEntitlementInfo = function(entityNames, successCallback
 };
 
+/**
+ * Get status for an Entity such as isSavable, progressPct, progressState. (Current only supports Collection)
```
progressPct doesn't match up with the native code in https://git.corp.adobe.com/dps/viewer-android/pull/1017 which uses downloadPercent.  The latter seems more descriptive.

mttmllns commented on www/CQMContext.js:
```diff
@@ -311,4 +323,94 @@ CQMContext.prototype._getEntitlementInfo = function(entityNames, successCallback
+/**
+ * Get status for an Entity such as isSavable, progressPct, progressState. (Current only supports Collection)
+ *
+ * @param {Entity} or {EntityList} Entity we want to get status for or EntityList we want to get status for.
```
You're missing the actual parameter name on some of the docs in this file (should it be "entityOrEntityList" here?)  "entity" and "isSilent" below, etc.

mcmuglim commented:
Hi @tmillett @adzellman, this is ready for maybe another quick review pass and would then be mergable. I added argchecks and updated README documentation.
